File: src/ast/BinaryExpression.java
package ast;

import java.io.StringWriter;

public class BinaryExpression extends BaseNode {
    private BaseNode _leftPart;
    private String _name;
    private BaseNode _rightPart;

    public BinaryExpression(BaseNode leftPart, String name, BaseNode rightPart) {
        super(NodeType.BinaryExpression);
        _leftPart = leftPart;
        _name = name;
        _rightPart = rightPart;
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        if (_name.equals(">")) {
            writer.write("(");
        }

        writer.write("(");
        _leftPart.Print(writer);
        writer.write(") ");

        writer.write(_name);

        writer.write(" (");
        _rightPart.Print(writer);
        writer.write(")");

        if (_name.equals(">")) {
            writer.write(")");
        }
    }
}

File: src/de/fearlesstobi/demangler/ast/BinaryExpression.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class BinaryExpression(private val leftPart: BaseNode, val _name: String, private val rightPart: BaseNode) : BaseNode(NodeType.BinaryExpression) {
    public override fun printLeft(writer: StringWriter) {
        if (_name == ">") {
            writer.write("(")
        }
        writer.write("(")
        leftPart.print(writer)
        writer.write(") ")
        writer.write(_name)
        writer.write(" (")
        rightPart.print(writer)
        writer.write(")")
        if (_name == ">") {
            writer.write(")")
        }
    }

    override fun getName(): String? {
        return _name
    }
}